service/services/services.go

```diff
@@ -413,8 +421,8 @@ func parseSpannerCondition(query *models.Query, pKey, sKey string) (string, map[
 func createWhereClause(whereClause string, expression string, queryVar string, RangeValueMap map[string]interface{}, params map[string]interface{}) (string, string) {
 	_, _, expression = utils.ParseBeginsWith(expression)
 	expression = strings.ReplaceAll(expression, "begins_with", "STARTS_WITH")
-
-	if whereClause != "WHERE " {
+	trimmedString := strings.TrimSpace(whereClause)
```

Comment:
You dont need new variable. This can be `whereClause := strings.TrimSpace(whereClause)`